src/framework/audio/internal/worker/mixer.cpp

```diff
@@ -153,39 +152,30 @@ std::shared_ptr<IMixerChannel> Mixer::channel(unsigned int number) const
     return m_inputList.at(number);
 }
 
-void Mixer::setBufferSize(unsigned int samples)
+void Mixer::forward(float* buffer, unsigned int sampleCount)
 {
     ONLY_AUDIO_WORKER_THREAD;
-    AbstractAudioSource::setBufferSize(samples);
-    for (auto& input : m_inputList) {
-        input.second->setBufferSize(samples);
-    }
-}
-
-void Mixer::forward(unsigned int sampleCount)
-{
-    ONLY_AUDIO_WORKER_THREAD;
-    std::fill(m_buffer.begin(), m_buffer.end(), 0.f);
 
     if (m_clock) {
         m_clock->forward(sampleCount);
     }
 
     for (auto& input : m_inputList) {
-        input.second->forward(sampleCount);
-        mixinChannel(input.second, sampleCount);
+        input.second->forward(buffer, sampleCount);
+        mixinChannel(buffer, input.second, sampleCount);
```

Comment:
I think, there needs one static buffer like in the `AudioBuffer::fillup()`

Reply:
actually, it's already a pointer to that write-cache buffer, not sure if whether we need another one here

Reply:
decided to add it after the discussion